src/Chainweb/Pact/SPV.hs

```diff
@@ -169,7 +177,77 @@ verifySPV bdb bh typ proof = runExceptT $ go typ proof
 
       t -> throwError $! "unsupported SPV types: " <> t
 
-
+-- | A tag for specifying the format of base64 error messages on chain.
+--
+--   `Legacy` errors match those produced by our legacy version of
+--   base64-bytestring, and are produced by parsing the error messages
+--   we receive from our current version of base64-bytestring (1.2),
+--   and formatting them in the older style. Legacy behavior also implies
+--   that non-canonical encodings be allowed, because that was the behavior
+--   of the legacy bytestring parser; and improperly padded messages
+--   will get extra padding, because that is the legacy chainweb behavior.
+--
+--   `Simplified` errors are a static string, which may not describe
+--   the issue with the base64-encoded string as well, but will be
+--   more stable as we upgrade base64 decoding libraries in the future.
+--   In the `Simplified` errors setting, messages rejected for using
+--   non-canonical encodings will remain as errors, because we want to
+--   begin enforcing the expected constraints on base64-encoded messages.
+--   Similarly, Simplified parsing will not add extra padding to improperly
+--   padded messages.
+data GenerateBase64ErrorMessage
```

Comment:
@emilypi and @edmundnoble here is a (verbose) strategy for forking the base64 decoding behavior. This code could use lots of scrutiny!